modules/GenericDataParser/src/server/generic_data_parser.R

```diff
@@ -577,12 +577,23 @@ validateCalculatedResults <- function(calculatedResults, dryRun, curveNames, tes
         rCompoundsDF <- batchProjectRestriced[batchProjectRestriced$isRestricted & batchProjectRestriced$Project.Code!=projectCode,]
         rCompounds <- rCompoundsDF$Requested.Name
         if (length(rCompounds) > 0) {
-          userRoles <- unlist(lapply(user$roles, function(role) role$roleEntry$roleName))
-          
-          if(configList$client.roles.crossProjectLoaderRole %in% userRoles) {
-            warnUser(paste0("Compounds from the following projects: ", paste(rCompoundsDF$name, collapse = "', '"),
-                            " will be loaded into project '",currentProj$name,"'"))
-          } else {
+          addProjectError <- TRUE
```

Comment:
Does this get set to FALSE if "shouldCheckRole" is false? I don't see an else block for the if(shouldCheckRole) block